kernel/src/main/java/org/kframework/compile/ExpandMacros.java

```diff
@@ -220,10 +222,14 @@ public K apply(KVariable k) {
            for (int i = 0; i < kapp.items().size(); i++) {
               final int idx = i;
               Set<Sort> sorts = sort(kapp.items().get(idx), r);
-              prods.removeIf(p -> sorts.stream().noneMatch(s -> mod.subsorts().lessThanEq(s, p.nonterminal(idx).sort())));
+              if (!sorts.contains(null)) {
```

Comment:
When would `sorts` be `null` and what would this function do in that case?

Reply:
Note that we return a sort of null if we examine the sort of a variable and it has no Sort attribute. This is done in order to make macro expansion work correctly when the user passes --non-strict to kompile. Basically, we're ignoring a sort check if the variable does not have a declared sort.